### PR TITLE
Add more retryable exceptions + cleanup

### LINKED ISSUE #429: Check retryable exceptions
There are some exceptions (i.e. 403, with reason "rateLimitExceeded") currently unlisted which may be retryable.  We should check best practices with the Cloud Resource Manager team.

### PR DESCRIPTION
**Pull request by ajkannan**
Closes #429

Included in the cleanup is a check for error code 404 not found when using "get" (see #440).

### CODE REVIEW

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
  */
 public interface ResourceManager extends Service<ResourceManagerOptions> {
 
-  public static final String DEFAULT_CONTENT_TYPE = "application/octet-stream";
+  String DEFAULT_CONTENT_TYPE = "application/octet-stream";
```
Comment (mziccard) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java]: I might be missing some background here but why did this become an instance variable?

Comment (aozarov) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java]: This is an interface.

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/spi/DefaultResourceManagerRpc.java [hunk 2]
```diff
@@ -82,7 +88,7 @@ public Project get(String projectId, Map<Option, ?> options) throws ResourceMana
           .execute();
     } catch (IOException ex) {
       ResourceManagerException translated = translate(ex);
-      if (translated.code() == HTTP_FORBIDDEN) {
+      if (translated.code() == HTTP_FORBIDDEN || translated.code() == HTTP_NOT_FOUND) {
```
Comment (mziccard) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/spi/DefaultResourceManagerRpc.java]: Just out of curiosity and related to #440: is 403 returned for both the not found and permission denied case? Is this done not to disclose info on existing projects?

Comment (ajkannan) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/spi/DefaultResourceManagerRpc.java]: Yup, 403 is returned for both not found and permission denied, and that is indeed the reason why.

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java]: Looks like one :)

**Pull request merged**